### ISSUE TITLE
Loading .ab1 files fails if the sample_id is not defined. Maybe define sample_id by default to some value?

### ISSUE THREAD

**Issue by ananswam** [biopython/biopython#1741]
### Setup

I am reporting a problem with Biopython version, Python version, and operating
system as follows:

```python
import sys; print(sys.version)
import platform; print(platform.python_implementation()); print(platform.platform())
import Bio; print(Bio.__version__)
```

3.5.5 |Anaconda custom (64-bit)| (default, Apr 26 2018, 08:11:22) 
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]
CPython
Darwin-15.4.0-x86_64-i386-64bit
1.70

### Expected behaviour

When reading in Sanger files (.ab1), I get an error that says local variable sequence_id referenced before assignment.

I was able to get around this by manually setting sample_id as follows in AbiIO.py:

```
if 'sample_id' not in locals():
    sample_id = 'NULL'
```

### Actual behaviour

This is what I was gettign before making a change.

```In [3]: a = SeqIO.read(open('EX25-1-A11-Sanger-For_A11.ab1','rb'),'abi')
---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
<ipython-input-3-c58003d62089> in <module>()
----> 1 a = SeqIO.read(open('EX25-1-A11-Sanger-For_A11.ab1','rb'),'abi')

~/anaconda3/lib/python3.5/site-packages/Bio/SeqIO/__init__.py in read(handle, format, alphabet)
    680     iterator = parse(handle, format, alphabet)
    681     try:
--> 682         first = next(iterator)
    683     except StopIteration:
    684         first = None

~/anaconda3/lib/python3.5/site-packages/Bio/SeqIO/__init__.py in parse(handle, format, alphabet)
    615             raise ValueError("Unknown format '%s'" % format)
    616         # This imposes some overhead... wait until we drop Python 2.4 to fix it
--> 617         for r in i:
    618             yield r
    619 

~/anaconda3/lib/python3.5/site-packages/Bio/SeqIO/AbiIO.py in AbiIterator(handle, alphabet, trim)
    424             file_name = ""
    425         record = SeqRecord(Seq(seq, alphabet),
--> 426                            id=sample_id, name=file_name,
    427                            description='',
    428                            annotations=annot,

UnboundLocalError: local variable 'sample_id' referenced before assignment
```

### Steps to reproduce

Try and load a ab1 file using the code above - i.e. 

```
from Bio import SeqIO
SeqIO.read(open('EX25-1-A11-Sanger-For_A11.ab1','rb'),'abi')
```

The error may not show up for all files.

**Comment by bow** [comment]
Hi @ananswam,

Thanks for reporting! This sounds like a case where a tag that we expected to be present in the file (SMPL1) is missing. Would it be possible to attach an example file using which the error was triggered? Also, do you know if something specific happened for those samples when their data was input to the machine? We expected the SMPL1 tag to be set by the sequencing machine operator, but in this case it may not have been the case or the sample ID value was set to other tags.

**Comment by ananswam** [comment]
Sure! Don't know what happened to the samples.
[EX25-1-A1-Sanger-For_A01.ab1.zip](https://github.com/biopython/biopython/files/2222598/EX25-1-A1-Sanger-For_A01.ab1.zip)

**Comment by bow** [comment]
Thanks @ananswam ~ the fix is really simple and they should be in `master` soon (see linked PR). In the meantime, if you would prefer to use an official release, your `locals()` fix should do the trick.

**Comment by bow** [comment]
Closing this as the fix has been merged in #1743. Feel free to reopen if you encounter this again.
